### PR TITLE
Python 3

### PR DESCRIPTION
**Pull request by buddly27**
Make the library compatible with Python 3.*

Closing #17.

### CODE REVIEW

**Comment by buddly27** [comment]
@kirijiri Ready to review

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 language: python
 python:
   - "2.7"
+  - "3.6"
+  - "3.7"
+  - "3.8"
```
Comment (kirijiri) [inline .travis.yml:6]: I guess it doesn't hurt to support more if it's easy enough, but really we'd only need 3.7, cause thats what the vfx reference platform recommends anyway: https://vfxplatform.com/

File: doc/release/release_notes.rst [hunk 2]
```diff
@@ -4,6 +4,28 @@
...
+        Added compatibility with python 3.6, 3.7 and 3.8.
+
+    .. change:: fixed
+
+        Fixed :class:`wiz.graph.Resolver` to ensure that conflicted nodes are
+        always sorted in ascending order of distance from the :attr:`root
+        <Graph.ROOT>` level of the graph.
+
+        Previously, conflicting nodes wouldn't be sorted properly when new
+        packages are added to the graph during the conflict resolution process.
```
Comment (kirijiri) [inline doc/release/release_notes.rst:20]: resulting in what? How did wiz ever work if this didn't work in the first place?

File: doc/release/release_notes.rst [hunk 3]
```diff
@@ -4,6 +4,28 @@
...
+
+    .. change:: fixed
+
+        Fixed :func:`wiz.utility.extract_version_ranges`.
```
Comment (kirijiri) [inline doc/release/release_notes.rst:24]: fixed what in it? The comment in the code says "Sort specifiers for deterministic results.", so

```suggestion
        Fixed :func:`wiz.utility.extract_version_ranges` to sort specifiers
        properly for deterministic results.
```
?

File: source/wiz/graph.py [hunk 4]
```diff
@@ -108,6 +108,15 @@ def __init__(self, definition_mapping):
...
         self._conflicts = collections.deque()
 
+        # Keep track of whether one or several packages are added to the graph
+        # during conflict resolution loop.
+
+        # Keep track of whether the conflict list needs to be sorted according
+        # to the latest distance mapping computed. It should always be true for
+        # the first conflict resolution loop, and be updated depending on
+        # whether the conflict list is updated.
+        self._conflicts_needs_sorting = True
```
Comment (kirijiri) [inline source/wiz/graph.py:118]: this is not part of any release note. what is this for? is this part of the version bug?

File: doc/release/release_notes.rst [hunk 5]
```diff
@@ -4,6 +4,28 @@
 Release Notes
 *************
 
+.. release:: Upcoming
+
+    .. change:: changed
+
+        Added compatibility with python 3.6, 3.7 and 3.8.
```
Comment (kirijiri) [inline doc/release/release_notes.rst:11]: I'd be more specific about what changes were introduced here, like:

```suggestion
        Updated :mod:`command_line`, :mod:`filesystem`, :mod:`package`, :mod:`system` 
        and :mod:`utility` to add compatibility with python 3.6, 3.7 and 3.8.
```

(unless the utility encoding change is unrelated to python3?)

**Comment by kirijiri** [comment]
Cloned branch into mill-wiz and built a python 3 wheel. Qip installed that wheel with all the python 3 requirements, so it could be run. Tested a couple of wiz commands and some wiz api calls, it seems to all work with python3!!!! well done!

File: .travis.yml (hunk 1 above)
Comment (buddly27) [reply .travis.yml:6]: It doesn't hurt to ensure that it works with future version. As long as the current standard version is in the tests..

File: doc/release/release_notes.rst (hunk 5 above)
Comment (buddly27) [reply doc/release/release_notes.rst]: Is is, thanks for pointing that out

File: doc/release/release_notes.rst (hunk 2 above)
Comment (buddly27) [reply doc/release/release_notes.rst:21]: Well it's a particular case when a new package is added during the conflict resolution process. See scenario 7 and 10 in the integration tests:

```
    Root
     |
     |--(A<=0.3.0): A==0.3.0
     |
     |--(B): B==0.1.0
     |   |
     |   `--(A !=0.3.0): A==1.0.0
     |
     `--(C): C==0.1.0
         |
         `--(A <1): A==0.9.0

    Expected: C==0.1.0, B==0.1.0, A==0.2.0
```

These test passed correctly in Python 2.7 because somehow the [conflict list concatenation](https://github.com/themill/wiz/blob/master/source/wiz/graph.py#L619) step produce surprisingly constant results...

File: source/wiz/graph.py (hunk 4 above)
Comment (buddly27) [reply source/wiz/graph.py:118]: This is part of the second release note. Do you think that much detail is needed in the notes?

File: .travis.yml (hunk 1 above)
Comment (buddly27) [reply .travis.yml]: 3.6 removed

File: doc/release/release_notes.rst (hunk 2 above)
Comment (buddly27) [reply doc/release/release_notes.rst]: Release notes updated
